src/varity/vcf_to_hgvs/protein.clj

```diff
@@ -80,38 +81,125 @@
+  (let [ter-site-positions (set (cond
+                                  (= strand :forward) (range (- cds-end 2) (inc cds-end))
+                                  (= strand :reverse) (range cds-start (inc (+ cds-start 2)))))
+        alt-positions (set (range pos (+ pos (count ref))))]
+    (boolean (seq (s/intersection ter-site-positions alt-positions)))))
+
+(defn- apply-offset
+  [pos ref alt exon-ranges ref-include-ter-site]
+  (letfn [(apply-offset* [exon-ranges*]
+            (ffirst (alt-exon-ranges exon-ranges* pos ref alt)))
+          (apply-ter-site-offset [pos*]
+            (-> (get-pos-exon-end-tuple pos* exon-ranges)
+                vector
+                apply-offset*))]
+    (fn [pos*]
+      (or (apply-offset* [[pos* pos*]])
+          (and ref-include-ter-site (apply-ter-site-offset pos*))
+          (some (fn [[_ e]] (when (<= e pos*) e)) (reverse exon-ranges))))))
+
 (defn- read-sequence-info
   [seq-rdr rg pos ref alt]
   (let [{:keys [chr tx-start tx-end cds-start cds-end exon-ranges strand]} rg
         ref-seq (cseq/read-sequence seq-rdr {:chr chr, :start cds-start, :end cds-end})
         alt-seq (common/alt-sequence ref-seq cds-start pos ref alt)
         alt-exon-ranges* (alt-exon-ranges exon-ranges pos ref alt)
-        ref-exon-seq1 (exon-sequence ref-seq cds-start exon-ranges)
-        ref-up-exon-seq1 (read-exon-sequence seq-rdr chr tx-start (dec cds-start) exon-ranges)
-        ref-up-exon-seq1 (subs ref-up-exon-seq1 (mod (count ref-up-exon-seq1) 3))
-        ref-down-exon-seq1 (read-exon-sequence seq-rdr chr (inc cds-end) tx-end exon-ranges)
-        nref-down-exon-seq1 (count ref-down-exon-seq1)
-        ref-down-exon-seq1 (subs ref-down-exon-seq1 0 (- nref-down-exon-seq1 (mod nref-down-exon-seq1 3)))
-        alt-exon-seq1 (exon-sequence alt-seq cds-start alt-exon-ranges*)
-        apply-offset #(or (ffirst (alt-exon-ranges [[% %]] pos ref alt))
-                          (some (fn [[_ e]] (when (<= e %) e)) (reverse alt-exon-ranges*)))]
-    {:ref-exon-seq ref-exon-seq1
-     :ref-prot-seq (codon/amino-acid-sequence (cond-> ref-exon-seq1
+        ref-exon-seq (exon-sequence ref-seq cds-start exon-ranges)
+        ref-up-exon-seq (read-exon-sequence seq-rdr chr tx-start (dec cds-start) exon-ranges)
+        alt-up-exon-seq (make-alt-up-exon-seq ref-up-exon-seq cds-start pos ref alt)
+        ref-down-exon-seq (read-exon-sequence seq-rdr chr (inc cds-end) tx-end exon-ranges)
+        alt-down-exon-seq (make-alt-down-exon-seq ref-down-exon-seq cds-start pos ref alt)
+        alt-exon-seq (exon-sequence alt-seq cds-start alt-exon-ranges*)
+        ter-site-adjusted-alt-seq (make-ter-site-adjusted-alt-seq alt-exon-seq alt-up-exon-seq alt-down-exon-seq
+                                                                  strand cds-start cds-end pos ref)
+        ref-include-ter-site (include-ter-site? rg pos ref)
+        apply-offset* (apply-offset pos ref alt alt-exon-ranges* ref-include-ter-site)]
```

Comment:
`apply-offset` and `apply-offset*` become more readable when you use the [`partial`](https://clojuredocs.org/clojure.core/partial) function.